pkg/controller/nodeipam/ipam/node_topology_syncer.go

```diff
@@ -3,28 +3,116 @@ package ipam
 import (
 	"context"
 	"fmt"
+	"strings"
+
 	nodetopologyv1 "github.com/GoogleCloudPlatform/gke-networking-api/apis/nodetopology/v1"
+	nodetopologyclientset "github.com/GoogleCloudPlatform/gke-networking-api/client/nodetopology/clientset/versioned"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	corelisters "k8s.io/client-go/listers/core/v1"
 	utilnode "k8s.io/cloud-provider-gcp/pkg/util/node"
+	"k8s.io/cloud-provider-gcp/providers/gce"
 	"k8s.io/klog/v2"
-	"strings"
 )
 
 const nodeTopologyCRName = "default"
 
-func (ca *cloudCIDRAllocator) updateNodeTopology(node *v1.Node) error {
-	ctx := context.Background()
+var (
+	// nodeTopologyReconciliationKey is a queue entry for nodeTopology reconciliation (used for node deletion).
+	nodeTopologyReconciliationKey = &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{Name: "nodeTopologyReconciliationKey"},
+	}
+)
+
+// NodeTopologySyncer processes nodetopology CR based on node add/update/delete events.
+type NodeTopologySyncer struct {
+	nodeTopologyClient nodetopologyclientset.Interface
+	cloud              *gce.Cloud
+	nodeLister         corelisters.NodeLister
+}
+
+func (syncer *NodeTopologySyncer) sync(node *v1.Node) error {
+	klog.V(0).Infof("syncing node topology CR for node item %v", node.GetName())
+	if node == nodeTopologyReconciliationKey {
```

Comment:
This seems too hacky, and yes better solutions do exist. I recall that usually, you'd do a nodeLister.Get() and check the error type returned is a NotFound which indicates the object has been deleted.

https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound